src/managers/GuildEventManager.js

```diff
@@ -0,0 +1,171 @@
+'use strict';
+
+const { Collection } = require('@discordjs/collection');
+const CachedManager = require('./CachedManager');
+const { TypeError, Error } = require('../errors');
+const GuildEvent = require('../structures/GuildEvent');
+const { PrivacyLevels, GuildEventEntityTypes } = require('../util/Constants');
+
+class GuildEventManager extends CachedManager {
+  constructor(guild, iterable) {
+    super(guild.client, GuildEvent, iterable);
+
+    /**
+     * The guild this manager belongs to
+     * @type {Guild}
+     */
+    this.guild = guild;
+  }
+
+  /**
+   * The cache of this manager
+   * @type {Collection<Snowflake, GuildEvent>}
+   * @name GuildEventManager#cache
+   */
+
+  /**
+   * Data that resolves to give a GuildEvent object. This can be:
+   * * A GuildEvent object
+   * * A Snowflake
+   * @typedef {GuildEvent|Snowflake} GuildEventResolvable
+   */
+
+  /**
+   * Options used to fetch a single guild event from a guild.
+   * @typedef {BaseFetchOptions} FetchGuildEventOptions
+   * @property {GuildEventResolvable} guildEvent The guild event to fetch
+   */
+
+  /**
+   * Options used to fetch multiple guild events from a guild.
+   * @typedef {Object} FetchGuildEventsOptions
+   * @property {boolean} [withUserCounts=true] Whether the number of users subscribed to the guild event
+   * should be returned
+   */
+
+  /**
+   * Obtains one or multiple guild events from Discord, or the guild cache if it's aready available.
+   * @param {GuildEventResolvable|FetchGuildEventOptions|FetchGuildEventsOptions} [options] The id of the guild event or
+   * options
+   * @returns {Promise<GuildEvent|Collection<Snowflake, GuildEvent>>}
+   */
+  async fetch(options = {}) {
+    const id = this.resolveId(options) ?? this.resolveId(options.guildEvent);
```

Comment:
```suggestion
    const id = this.resolveId(options.guildEvent ?? options);
```
this would probably make more sense

Reply:
You’d need to add nullish coalescing then because options might be a snowflake

Reply:
a snowflake isn't null/undefined

Reply:
Ah nvm I thought that would throw an error but I just confused it for sth else